hatch_pip_compile/plugin.py

```diff
@@ -70,39 +85,42 @@ def get_option_types() -> Dict[str, Any]:
             "pip-compile-hashes": bool,
             "pip-compile-args": List[str],
             "pip-compile-constraint": str,
+            "pip-compile-installer": str,
+            "pip-compile-install-args": List[str],
         }
 
-    def _hatch_pip_compile_install(self):
+    def install_pip_tools(self) -> None:
+        """
+        Install pip-tools (if not already installed)
         """
-        Run the full hatch-pip-compile install process
-
-        1) Create virtual environment if not exists
-        2) Install pip-tools
-        3) Run pip-compile (if lock file does not exist / is out of date)
-        4) Run pip-sync
-        5) Install project in dev mode
-        """
-        try:
-            _ = self.virtual_env.executables_directory
-        except OSError:
-            self.create()
         with self.safe_activation():
-            self.virtual_env.platform.check_command(
-                self.construct_pip_install_command(["pip-tools"])
+            in_sync = dependencies_in_sync(
+                requirements=[Requirement("pip-tools")],
+                sys_path=self.virtual_env.sys_path,
+                environment=self.virtual_env.environment,
             )
-            if not self.lockfile_up_to_date:
-                self._pip_compile_cli()
-            self._pip_sync_cli()
-        if not self.skip_install:
-            if self.dev_mode:
-                super().install_project_dev_mode()
-            else:
-                super().install_project()
+            if not in_sync:
+                self.platform.check_command(self.construct_pip_install_command(["pip-tools"]))
 
-    def _pip_compile_cli(self) -> None:
+    def run_pip_compile(self) -> None:
+        """
+        Run pip-compile if necessary
+        """
+        if not self.lockfile_up_to_date:
+            self.install_pip_tools()
+            if self.piptools_lock_file.exists():
+                _ = self.piptools_lock.compare_python_versions(
+                    verbose=self.config.get("pip-compile-verbose", None)
+                )
+            self.pip_compile_cli()
```

Comment:
There is a `safe_activation` context needed here

```suggestion
        if not self.lockfile_up_to_date:
            self.install_pip_tools()
            with self.safe_activation():
                if self.piptools_lock_file.exists():
                    _ = self.piptools_lock.compare_python_versions(
                        verbose=self.config.get("pip-compile-verbose", None)
                    )
                self.pip_compile_cli()
```